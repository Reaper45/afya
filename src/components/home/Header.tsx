import React from "react";

// Local Files
import styled from "lib/emotion";
import Button from "components/common/Button";
import Progress from "components/common/Progress";
import Avatar from "components/common/Avatar";

import avatar from "static/avatar.jpg";

const HeaderWrapper = styled("header")`
  display: flex;
  background: #fff;
  padding: 1.5rem 2em;
  justify-content: space-between;
  align-items: center;
`;

const RightSection = styled("div")`
  display: flex;
  align-items: center;
  > span:first-of-type {
    width: 160px;
  }
  > div:last-of-type {
    margin-left: 3em;
  }
`;

const Header: React.FC = () => {
  return (
    <HeaderWrapper>
      <Button>Add Assessment +</Button>
      <RightSection>
        <Progress
          inverted
          progress={40}
          description="24k / 40k patients seen"
        />
        <Avatar
          img={avatar}
          title="Sandra Mamai"
          description="sandra.mamai@test.com"
        />
      </RightSection>
    </HeaderWrapper>
  );
};

export default Header;
