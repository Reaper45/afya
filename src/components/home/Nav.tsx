import React from "react";
import { NavLink } from "react-router-dom";

// Local Files
import styled from "lib/emotion";
import Icon, { chatIcon, folderIcon, gridIcon, graphIcon } from "components/common/Icon";

const NavWrapper = styled("nav")`
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 0 2rem;
  background: #fff;
  border-right: 1px solid ${props => props.theme.colors.grey};
  box-sizing: border-content;
  position: fixed;
  top: 0;
  left: 0;
  a {
    padding: 1rem 0;
    svg {
      fill: ${props => props.theme.colors.accent};
      opacity: .33;
    }
    &.active {
      svg {
        fill: ${props => props.theme.colors.primary};
        opacity: 1;
      }
    }
  }
`;

const Nav: React.FC = () => {
  return (
    <NavWrapper>
      <NavLink to="/files" exact>
        <Icon icon={folderIcon}/>
      </NavLink>
      <NavLink to="/" exact>
        <Icon icon={graphIcon}/>
      </NavLink>
      <NavLink to="/inbox" exact>
        <Icon icon={chatIcon}/>
      </NavLink>
      <NavLink to="/settings" exact>
        <Icon icon={gridIcon}/>
      </NavLink>
    </NavWrapper>
  );
}

export default Nav
