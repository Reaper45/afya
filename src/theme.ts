export interface ITheme {
  colors: {
    primary: string;
    dark: string;
    light: string;
    secondary: string;
    accent: string;
  };
}

export const theme: ITheme = {
  colors: {
    primary: "#5A67D8",
    dark: "#3C366B",
    light: "#A3BFFA",
    secondary: "#222222",
    accent: "A0AEC0"
  }
};
