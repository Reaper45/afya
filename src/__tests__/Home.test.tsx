import React from "react";
import { shallow, mount } from "enzyme";
import { ThemeProvider } from "emotion-theming";
import { Router } from "react-router-dom";
import { createBrowserHistory } from "history";

import { theme } from "theme";

import Home from "components/home";
import Nav from "components/home/Nav";
import Header from "components/home/Header";

const history = createBrowserHistory();

describe("Test correct home layout", () => {
  it("renders without crashing", () => {
    const component = shallow(<Home />);
    expect(component).toMatchSnapshot();
  });

  it("should contain the nav bar", () => {
    const component = shallow(<Home />);
    expect(component.containsMatchingElement(<Nav />));
  });

  it("should container the main content section", () => {
    const component = shallow(<Home />);
    expect(component.containsMatchingElement(<main />));
  });

  it("should contain the header", () => {
    const component = shallow(<Home />);
    expect(component.containsMatchingElement(<Header />));
  });
});

it("renders visits correctly", () => {
  const component = mount(
    <ThemeProvider theme={theme}>
      <Router history={history}>
        <Home />
      </Router>
    </ThemeProvider>
  );
  expect(
    component.find('[data-name="visit-items"]').find("button")
  ).toHaveLength(6);
});


it("renders issues correctly", () => {
  const component = mount(
    <ThemeProvider theme={theme}>
      <Router history={history}>
        <Home />
      </Router>
    </ThemeProvider>
  );
  expect(component.find('[data-name="issues"]').find("button")).toHaveLength(6);
});
