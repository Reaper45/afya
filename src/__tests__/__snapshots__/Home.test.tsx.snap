// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test correct home layout renders without crashing 1`] = `
<Styled(div)>
  <Nav />
  <Styled(div)>
    <Header />
    <Styled(main)>
      <Styled(div)>
        Analytics
      </Styled(div)>
      <Styled(div)>
        <Styled(div)>
          <b>
            Visits
          </b>
          <br />
          <FilterItem
            active={true}
            count={7}
            key="1"
            onClick={[Function]}
            title="Kiambiu"
          />
          <FilterItem
            active={false}
            count={12}
            key="2"
            onClick={[Function]}
            title="Mkuru Kwa Ruben"
          />
          <FilterItem
            active={false}
            count={26}
            key="3"
            onClick={[Function]}
            title="Mkuru Kwa Njenga"
          />
          <FilterItem
            active={false}
            count={38}
            key="4"
            onClick={[Function]}
            title="Baba dogo"
          />
          <FilterItem
            active={false}
            count={41}
            key="5"
            onClick={[Function]}
            title="Kosovo"
          />
          <FilterItem
            active={false}
            count={24}
            key="6"
            onClick={[Function]}
            title="Mukuru Kayaba"
          />
        </Styled(div)>
      </Styled(div)>
    </Styled(main)>
  </Styled(div)>
</Styled(div)>
`;
